controllers/applicationvolumereplication_controller.go

```diff
@@ -153,20 +176,78 @@ func BuildManifestWorkName(name, namespace, mwType string) string {
 	return fmt.Sprintf(ManifestWorkNameFormat, name, namespace, mwType)
 }
 
+func ManifestWorkPredicateFunc() predicate.Funcs {
+	mwPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log := ctrl.Log.WithName("ManifestWork")
+
+			oldMW, ok := e.ObjectOld.DeepCopyObject().(*ocmworkv1.ManifestWork)
+			if !ok {
+				log.Info("Failed to deep copy older ManifestWork")
+
+				return false
+			}
+			newMW, ok := e.ObjectNew.DeepCopyObject().(*ocmworkv1.ManifestWork)
+			if !ok {
+				log.Info("Failed to deep copy newer ManifestWork")
+
+				return false
+			}
+
+			log.Info("Update event for ManifestWork")
+
+			return !reflect.DeepEqual(oldMW.Spec, newMW.Spec)
+		},
+	}
+
+	return mwPredicate
+}
+
+func filterMW(mw *ocmworkv1.ManifestWork) []ctrl.Request {
+	return []ctrl.Request{
+		reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      mw.Annotations[AVRNameAnnotation],
+				Namespace: mw.Annotations[AVRNamespaceAnnotation],
+			},
+		},
+	}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *ApplicationVolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	pred := predicate.GenerationChangedPredicate{}
+	genChangedPred := predicate.GenerationChangedPredicate{}
+	mwPred := ManifestWorkPredicateFunc()
+
+	mwMapFun := handler.EnqueueRequestsFromMapFunc(handler.MapFunc(func(obj client.Object) []reconcile.Request {
+		mw, ok := obj.(*ocmworkv1.ManifestWork)
+		if !ok {
+			ctrl.Log.Info("ManifestWork map function received non-MW resource")
+
+			return []reconcile.Request{}
+		}
+
+		// ctrl.Log.Info(fmt.Sprintf("MapFunc for ManifestWork (%v)", mw))
+		return filterMW(mw)
+	}))
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&rmn.ApplicationVolumeReplication{}).
-		WithEventFilter(pred).
+		WithEventFilter(genChangedPred).
+		Watches(&source.Kind{Type: &ocmworkv1.ManifestWork{}}, mwMapFun, builder.WithPredicates(mwPred)).
```

Comment:
(question) Like VRG owns VRs it creates, and also sets up a SetControllerReference, can we not own created MWs?

Checked some ACM [code](https://github.com/open-cluster-management/work/blob/11688af42ddcf9911ea68b348974fabb684e5e2e/pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go#L144-L148), it is possible that MWApplied is set as the owner, and we can have only one owner, and so this may not be feasible (although this seems to be the spoke reconciler).

The advantage is we do not get woken up for any MW (which is achieve using filters for now), and the appropriate AVR is reconciled without storing it in the annotation, if we can achieve the above.